.github/workflows/docker-build-push.yml

```diff
@@ -2,29 +2,31 @@ name: Publish Docker image
 
 on:
   push:
-    branches:
-      - master
+    tags:
+      - '*'
 
 jobs:
   publish-docker:
 
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
 
-    - uses: AdoptOpenJDK/install-jdk@v1
-      with:
-        version: '14'
-        architecture: x64
-    - name: Build with Maven
-      run: ./mvnw clean package
+      - uses: AdoptOpenJDK/install-jdk@v1
+        with:
+          version: '14'
+          architecture: x64
+      - name: Build with Maven (no tests)
+        run: ./mvnw clean package -DskipTests
 
-    - name: Build and push Docker images
-      uses: docker/build-push-action@v1
-      with:
-        username: vthakurdocker
-        password: ${{ secrets.DOCKER_PASSWORD }}
-        repository: vthakurdocker/pitchfork
-        tags: latest_gh
+      - name: Build and push Docker images on new git tags
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+          repository: expediagroup/pitchfork
+          tag_with_ref: true
```

Comment:
this will also push on branch, too, iiuc https://github.com/docker/build-push-action#tag_with_ref

Reply:
Maybe I'm understanding the docs wrong but I think it will only push for new git tags.

```tag_with_ref: (...) For git tags the reference will be refs/tags/{git-tag} and the tag will be {git-tag}.```:

And there's these restrictions to just push on the git tags.

```
on:
  push:
    tags:
      - '*'
```

and:

```
The following will only push the image when the event that kicked off the workflow was a push of a git tag:

push: ${{ startsWith(github.ref, 'refs/tags/') }}
```

Reply:
Thanks for teaching me! I believe it!

Reply:
I tested this in my fork and it seems to work as expected. Fingers crossed 🤞